server/broadcast.go

```diff
@@ -473,34 +479,57 @@ func processSegment(cxn *rtmpConnection, seg *stream.HLSSegment) ([]string, erro
 		sv = verification.NewSegmentVerifier(Policy)
 	}
 
-	for i := 0; i < MaxAttempts; i++ {
-		// if fails, retry; rudimentary
-		var urls []string
-		if urls, err = transcodeSegment(cxn, seg, name, sv); err == nil {
-			return urls, nil
+	var (
+		startTime = time.Now()
+		attempts  []data.TranscodeAttemptInfo
+		urls      []string
+	)
+	for len(attempts) < MaxAttempts {
+		// if transcodeSegment fails, retry; rudimentary
+		var info data.TranscodeAttemptInfo
+		urls, info, err = transcodeSegment(cxn, seg, name, sv)
+		attempts = append(attempts, info)
+		if err == nil {
+			break
 		}
 
 		if shouldStopStream(err) {
 			glog.Warningf("Stopping current stream due to err=%v", err)
 			rtmpStrm.Close()
-			return nil, err
+			break
 		}
-
 		if isNonRetryableError(err) {
 			glog.Warningf("Not retrying current segment nonce=%d seqNo=%d due to non-retryable error err=%v", nonce, seg.SeqNo, err)
-			return nil, err
+			break
 		}
-
 		// recoverable error, retry
 	}
-	if err != nil {
+	if MetadataQueue != nil {
+		success := err == nil && len(urls) > 0
+		key, evt := newTranscodeEvent(mid, seg, startTime, success, attempts)
+		go func() {
+			ctx, cancel := context.WithTimeout(context.Background(), MetadataPublishTimeout)
+			defer cancel()
+			if err := MetadataQueue.Publish(ctx, key, evt, false); err != nil {
+				glog.Errorf("Error publishing stream transcode event: err=%q, manifestId=%q, seqNo=%d, key=%q, event=%+v", err, mid, seg.SeqNo, key, evt)
```

Comment:
```suggestion
				glog.Errorf("Error publishing stream transcode event: err=%q, manifestID=%q, seqNo=%d, key=%q, event=%+v", err, mid, seg.SeqNo, key, evt)
```

Since `manifestID` is used as the key elsewhere in logging.

Also, minor note: I don't think we use a comma + space as the delimiter for the key=value pairs in logging elsewhere so for consistency we could just stick with a space as the delimiter i.e. `manifestID=%q seqNo=%d ...`.